src/shims/unix/sync.rs

```diff
@@ -704,7 +724,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
-        if !matches!(&*this.tcx.sess.target.os, "linux" | "macos") {
+        if !matches!(&*this.tcx.sess.target.os, "linux" | "macos" | "solaris" | "illumos") {
```

Comment:
You'll need to do the same in pthread_condattr_setclock, pthread_condattr_getclock.

(Also I just realized the function name there is wrong... while at it, could you fix it?)